Packs/Syslog/Integrations/SyslogSender/SyslogSender.py

```diff
@@ -54,60 +61,135 @@
+            )
+            # Connect to the syslog server
+            self.socket.send(syslog_message.encode('utf-8'))
+
+        except Exception as e:
+            if self.socket:
+                self.socket.close()
+            demisto.error(str(e))
+
 
 ''' Syslog Manager '''
 
 
 class SyslogManager:
-    def __init__(self, address: str, port: int, protocol: str, logging_level: int, facility: int):
+    def __init__(self, address: str, port: int, protocol: str, logging_level: int,
+                 facility: int, cert_path: str | None, self_signed_certificate: bool):
         """
         Class for managing instances of a syslog logger.
         :param address: The IP address of the syslog server.
         :param port: The port of the syslog server.
-        :param protocol: The messaging protocol (TCP / UDP).
+        :param protocol: The messaging protocol (TCP / UDP / TLS).
         :param logging_level: The logging level.
         """
         self.address = address
         self.port = port
         self.protocol = protocol
         self.logging_level = logging_level
         self.facility = facility
+        self.syslog_cert_path = cert_path
+        self.self_signed_cert = self_signed_certificate
 
     @contextmanager  # type: ignore[misc, arg-type]
     def get_logger(self) -> Generator:
         """
         Get a new instance of a syslog logger.
         :return: syslog logger
         """
-        handler = self._get_handler()
+        if self.protocol == TLS and self.syslog_cert_path:
+            handler = self.init_handler_tls(self.syslog_cert_path)
+            demisto.debug('get handler tls')
+        else:
+            handler = self._get_handler()
         syslog_logger = self._init_logger(handler)
+        demisto.debug('logger was created ')
         try:
             yield syslog_logger
         finally:
             syslog_logger.removeHandler(handler)
             handler.close()
 
-    def _get_handler(self) -> SysLogHandler:
-        """
-        Get a syslog handler for a logger according to provided parameters.
-        :return: syslog handler
-        """
-        address: Union[str, Tuple[str, int]] = (self.address, self.port)
-        kwargs: Dict[str, Any] = {
-            'facility': self.facility
-        }
-
-        if self.protocol == TCP:
-            kwargs['socktype'] = SOCK_STREAM
-        elif self.protocol == 'unix':
-            address = self.address
-
-        kwargs['address'] = address
-
-        return SysLogHandler(**kwargs)
-
-    def _init_logger(self, handler: SysLogHandler) -> Logger:
+    def _get_handler(self) -> Rfc5424SysLogHandler:
+        sock_kind = SOCK_STREAM if self.protocol == TCP else socket.SOCK_DGRAM
+        return Rfc5424SysLogHandler(address=(self.address, self.port),
+                                    facility=self.facility,
+                                    socktype=sock_kind,
+                                    timeout=10,
```

Comment:
Why 10?
What's the default timeout if you leave it empty?

Reply:
5, should I remove it then?

Reply:
I removed it